test/flags-preload.js

```diff
@@ -0,0 +1,164 @@
+    ].join(' '), cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(sliceLines(stdout, 0, 1)).toEqual('Preloading external module: ../test-module.js');
+      expect(sliceLines(stdout, 1, 2)).toEqual('inside test module');
+      expect(sliceLines(stdout, 2, 3)).toEqual('Preloaded external module: ../test-module.js');
+
+      var chgWorkdirLog = sliceLines(stdout, 3, 4);
+      var workdir = 'test/fixtures/gulpfiles'.replace(/\//g, path.sep);
+      expect(chgWorkdirLog).toMatch('Working directory changed to ');
+      expect(chgWorkdirLog).toMatch(workdir);
+
+      stdout = sliceLines(stdout, 5);
+      expect(stdout).toEqual(
+        'Starting \'default\'...\n' +
+         'Starting \'test1\'...\n' +
+          'Starting \'noop\'...\n' +
+          'Finished \'noop\' after ?\n' +
+         'Finished \'test1\' after ?\n' +
+         'Starting \'test3\'...\n' +
+          'Starting \'described\'...\n' +
+          'Finished \'described\' after ?\n' +
+         'Finished \'test3\' after ?\n' +
+         'Starting \'noop\'...\n' +
+         'Finished \'noop\' after ?\n' +
+        'Finished \'default\' after ?\n' +
+        ''
+      );
+      done(err);
+    }
+  });
+
+  it('can preload multiple modules before running gulpfile', function(done) {
+    exec([
+      'cd ' + baseDir + cmdSep,
+      gulpCmd,
+      '--preload ../test-module.js',
+      '--preload ../test-module-2.js',
+      '--cwd ./test/fixtures/gulpfiles',
+    ].join(' '), cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(sliceLines(stdout, 0, 1)).toEqual('Preloading external module: ../test-module.js');
+      expect(sliceLines(stdout, 1, 2)).toEqual('inside test module');
+      expect(sliceLines(stdout, 2, 3)).toEqual('Preloaded external module: ../test-module.js');
+      expect(sliceLines(stdout, 3, 4)).toEqual('Preloading external module: ../test-module-2.js');
+      expect(sliceLines(stdout, 4, 5)).toEqual('inside test module 2');
+      expect(sliceLines(stdout, 5, 6)).toEqual('Preloaded external module: ../test-module-2.js');
+      done(err);
+    }
+  });
+
+  it('warns if module doesn\'t exist', function(done) {
+    exec([
+      'cd ' + baseDir + cmdSep,
+      gulpCmd,
+      '--preload ./null-module.js',
+      '--cwd ./test/fixtures/gulpfiles',
+    ].join(' '), cb);
+
+    function cb(err, stdout, stderr) {
+      expect(err).toBeNull();
+      expect(stderr).toEqual('');
+      expect(sliceLines(stdout, 0, 2)).toEqual(
+        'Preloading external module: ./null-module.js\n' +
+        'Failed to preload external module: ./null-module.js'
+      );
+      expect(sliceLines(stdout, 2, 3)).toMatch('Error: Cannot find module \'./null-module.js\'');
+      expect(stdout).not.toMatch('inside test module');
+      expect(stdout).not.toMatch('Preloaded external module: ../null-module.js');
```

Comment:
```suggestion
      expect(stdout).not.toMatch('Preloaded external module: ./null-module.js');
```
Would this be `./null-module.js`?

Reply:
Do we even need these `.not.toMatch` expectations? They really confused me.

Reply:
I'll remove these `not.toMatch`s.